src/opencl/image_function_opencl.cpp

```diff
@@ -882,6 +894,39 @@ namespace Image_Function_OpenCL
         multiCL::launchKernel2D( kernel, width, height );
     }
 
+    bool IsEqual( const Image & in1, const Image & in2 )
+    {
+        Image_Function::ParameterValidation( in1, in2 );
+
+        return Image_Function_Helper::IsEqual( IsEqual, in1, in2 );
+    }
+
+    bool IsEqual( const Image & in, uint32_t startXIn, uint32_t startYIn, const Image & out, uint32_t startXOut, uint32_t startYOut,
+                 uint32_t width, uint32_t height )
+    {
+        Image_Function::ParameterValidation( in, startXIn, startYIn, out, startXOut, startYOut, width, height );
+
+        const multiCL::OpenCLProgram & program = GetProgram();
+        multiCL::OpenCLKernel kernel( program, "isEqualOpenCL");
+
+        const uint8_t colorCount = Image_Function::CommonColorCount( in, out );
+        width = width * colorCount;
+
+        const uint32_t rowSizeIn  = in.rowSize();
+        const uint32_t rowSizeOut = out.rowSize();
+
+        const uint32_t offsetIn  = startYIn  * rowSizeIn  + startXIn  * colorCount;
+        const uint32_t offsetOut = startYOut * rowSizeOut + startXOut * colorCount;
+
+        uint32_t equal = 0;
```

Comment:
You cannot use CPU allocated variable for GPU kernels. Please use `multiCL::Type< uint32_t > differenceCount( 0u );` then pass into kernel as `differenceCount.data()` and compare it as `differenceCount.get() == 0`.